### PR TITLE
LAB-4040 Get model from platform

### PR DESCRIPTION
**Pull request by roikoren755**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -24,7 +24,8 @@ pyparsing==2.4.5
 einops==0.3.2
 pycocotools==2.0.4
 protobuf~=3.19.0
-deci-lab-client==2.38.0
+deci-common==3.4.1
+deci-lab-client==2.55.0b16158
```
Comment (ofrimasad) [inline requirements.txt:28]: please use a released version

File: src/super_gradients/common/plugins/deci_client.py [hunk 2]
```diff
@@ -1,24 +1,42 @@
...
+    def __init__(self, lab_client: DeciPlatformClient):
+        self.lab_client = lab_client
         GlobalHydra.instance().clear()
+        self.super_gradients_version = None
+        try:
+            import pkg_resources
+
+            self.super_gradients_version = pkg_resources.get_distribution("super_gradients").version
+        except Exception as e:
+            print(e)
```
Comment (ofrimasad) [inline src/super_gradients/common/plugins/deci_client.py:21]: printing won't do any good. 
instead add `self.super_gradients_version = 3.0.0` as a fallback

File: src/super_gradients/common/plugins/deci_client.py [hunk 3]
```diff
@@ -1,24 +1,42 @@
...
+except (ImportError, NameError):
+    from typing import Any as DeciPlatformClient
 
-    def __init__(self):
-        self.bucket_name = 'deci-model-repository-research'
-        self.prefix = 'mock_client'
-        self.s3 = boto3.client('s3')
+
+class DeciClient:
+    def __init__(self, lab_client: DeciPlatformClient):
```
Comment (ofrimasad) [inline src/super_gradients/common/plugins/deci_client.py:12]: why not instantiate the DeciPlatformClient inside? why do we need to pass it all the way from outside?

**Review by ofrimasad — COMMENTED** [review]
my comment inline

File: requirements.txt (hunk 1 above)
Comment (roikoren755) [reply requirements.txt:28]: yes, just waiting on a new version of the API to be released

File: src/super_gradients/common/plugins/deci_client.py (hunk 2 above)
Comment (roikoren755) [reply src/super_gradients/common/plugins/deci_client.py:21]: 👍

File: src/super_gradients/common/plugins/deci_client.py (hunk 3 above)
Comment (roikoren755) [reply src/super_gradients/common/plugins/deci_client.py:12]: The lab client needs to be logged in for anything to work.
With `deci-adk`, we provide colabs and snippets that initialize the client and log in with it first, and then pass it to the class to fetch the model using it

Comment (ofrimasad) [reply src/super_gradients/common/plugins/deci_client.py:12]: but login can happen on `DeciPlatformClient.__init__()`. it's not like we expect the user to pass a user name or token in the code. these are taken from env_variable or local file (like with was cli)

Comment (roikoren755) [reply src/super_gradients/common/plugins/deci_client.py:16]: The snippets we currently have do expect the user to pass a token in the code, though

Comment (ofrimasad) [reply src/super_gradients/common/plugins/deci_client.py:16]: snippets are ok, but real code does not include tokens.
you can have an optional parameter `DeciPlatformClient.__init__(self, user: Optional[str], token: Optional[str])` but the DeciPlatformClient needs to know how to init without it (like wandb, was and other clients).
either take the token from env/file or prompt for login

Comment (roikoren755) [reply src/super_gradients/common/plugins/deci_client.py]: 👍

**Comment by ofrimasad** [comment]
moved to other PR

**Pull request closed without merging**